src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp

```diff
@@ -382,7 +382,7 @@ class boss_blindeye_the_seer : public CreatureScript
             me->SetInCombatWithZone();
             instance->SetBossState(DATA_MAULGAR, IN_PROGRESS);
 
-            events.ScheduleEvent(EVENT_ADD_ABILITY1, 5000);
+            events.ScheduleEvent(EVENT_ADD_ABILITY1, 1);
```

Comment:
Uneducated guess, but wouldn't this mean that the ability is cast within 1ms of combat starting?

Reply:
yes, and I have yet to find a video where the PW:S is casted practically on pull

Reply:
I understand the issue as "Spell Shield buff as first cast on pull" means it should be cast on pull?

Reply:
most of the discussion is in the initial ticket https://github.com/chromiecraft/chromiecraft/issues/4811
long story short, the cast should occur within 1-2 seconds of the pull - but be the first spell casted (must happen before the first fireball and blastnova
see also
https://github.com/chromiecraft/chromiecraft/issues/4873

Reply:
1.2s if wanting to replicate Blizz queueing and packet cycles

Reply:
But honestly that whole part is a mess. Tying 4 creatures' spells together. Scripting made by insane people that is.

Reply:
I can fix it that way.

They are not tied together tho, just named the same x)